### PR TITLE
Google Pay improvements

### PR DESCRIPTION
**Pull request by OscarSpruit**
## Description
This PR is a huge compilation of smaller PRs. These small PRs have all been reviewed.

## Checklist 
- [x] PR is labelled 
- [x] Code is unit tested
- [x] Changes are tested manually
- [x] Related issues are linked

COAND-941

### CODE REVIEW

File: googlepay/src/main/java/com/adyen/checkout/googlepay/GooglePayComponent.kt [hunk 1]
```diff
@@ -74,10 +83,27 @@ class GooglePayComponent internal constructor(
      * @param activity    The activity to start the screen and later receive the result.
      * @param requestCode The code that will be returned on the [Activity.onActivityResult]
      */
+    @Deprecated("Deprecated in favor of startGooglePayScreen()", ReplaceWith("submit()"))
```
Comment (dmengelt) [inline googlepay/src/main/java/com/adyen/checkout/googlepay/GooglePayComponent.kt:86]: Deprecated in favor of `submit()` I guess?

Comment (OscarSpruit) [reply googlepay/src/main/java/com/adyen/checkout/googlepay/GooglePayComponent.kt:86]: Good catch 🦅

File: RELEASE_NOTES.md [hunk 2]
```diff
@@ -9,68 +9,23 @@
...
-
-  | Locale     | Values    |
-  |------------|-----------|
-  | Catalan    | **ca-ES** |
-  | Icelandic  | **is-IS** |
-  | Bulgarian  | **bg-BG** |
-  | Estonian   | **et-EE** |
-  | Latvian    | **lv-LV** |
-  | Lithuanian | **lt-lT** |
+- Launch Google Pay with `submit()` to get rid of the deprecated activity result handling.
```
Comment (jreij) [inline RELEASE_NOTES.md:12]: WDYT of adding a migration document from the old to the new flow?

File: googlepay/src/main/java/com/adyen/checkout/googlepay/internal/ui/DefaultGooglePayDelegate.kt [hunk 3]
```diff
@@ -106,6 +108,7 @@ internal class DefaultGooglePayDelegate(
             paymentMethod,
             componentParams,
         ) { isAvailable, _ ->
+            this.isAvailable = isAvailable
```
Comment (araratthehero) [inline googlepay/src/main/java/com/adyen/checkout/googlepay/internal/ui/DefaultGooglePayDelegate.kt:111]: What do you think of storing this in the `outputData`, similar to how we store `paymentData`?

Comment (OscarSpruit) [reply googlepay/src/main/java/com/adyen/checkout/googlepay/internal/ui/DefaultGooglePayDelegate.kt:111]: `outputData` is generally used to communicate with the view, so I'm not sure if it makes sense to store state in there 🤔

Comment (araratthehero) [reply googlepay/src/main/java/com/adyen/checkout/googlepay/internal/ui/DefaultGooglePayDelegate.kt:111]: I get that point. In that case I guess `paymentData` should also be removed from there and it is not part of this task. Let's leave it as it is then.

**Review by araratthehero — APPROVED** [review]

**Review by jreij — CHANGES_REQUESTED** [review]
Adding request changes to block merging this, in case we need to postpone it a bit

**Review by jreij — COMMENTED** [review]
Can be reviewed and merged when ready

File: RELEASE_NOTES.md
Comment (jreij) [inline RELEASE_NOTES.md:1]: The formatting is broken here

**Review by jreij — APPROVED** [review]
🎉

File: RELEASE_NOTES.md
Comment (jreij) [inline RELEASE_NOTES.md:1]: The formatting is broken here

**Review by jreij — CHANGES_REQUESTED** [review]
Release notes should be fixes

**Review by jreij — APPROVED** [review]

**Pull request merged**